pkg/health/health.go

```diff
@@ -0,0 +1,163 @@
+		if _, okay := nodeList[nodeName]; !okay {
+			nodeList[nodeName] = podList.Items[i].Name
+		}
+	}
+
+	if len(nodeList) < 3 {
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes", daemonType)
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	cephHealthDetails := exec.RunCommandInOperatorPod(context, "ceph", []string{"health", "detail"}, operatorNamespace, clusterNamespace, false)
+	if strings.Contains(cephHealthDetails, "HEALTH_OK") {
+		fmt.Printf("\n%s", cephHealthDetails)
+	} else if strings.Contains(cephHealthDetails, "HEALTH_WARN") {
+		fmt.Println("Warning: ", cephHealthDetails)
+	} else if strings.Contains(cephHealthDetails, "HEALTH_ERR") {
+		fmt.Println("Error: ", cephHealthDetails)
+	}
+}
+
+func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	operatorNamespacePods, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", operatorNamespace, err)
+	}
+
+	clusterNamespacePods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", clusterNamespace, err)
+	}
+
+	var podNotRunning, podRunning []v1.Pod
+
+	for i := range operatorNamespacePods.Items {
+		if operatorNamespacePods.Items[i].Status.Phase != v1.PodRunning {
+			podNotRunning = append(podNotRunning, operatorNamespacePods.Items[i])
+		} else {
+			podRunning = append(podRunning, operatorNamespacePods.Items[i])
+		}
+	}
+
+	for i := range clusterNamespacePods.Items {
+		if operatorNamespacePods.Items[i].Status.Phase != v1.PodRunning {
+			podNotRunning = append(podNotRunning, operatorNamespacePods.Items[i])
+		} else {
+			podRunning = append(podRunning, operatorNamespacePods.Items[i])
+		}
+	}
+
+	fmt.Println("Info: Pods that are in 'Running' status")
+	for i := range podRunning {
+		fmt.Println(podRunning[i].Name, "\t", podRunning[i].Status.Phase, "\t", podRunning[i].Namespace, "\t", podRunning[i].Spec.NodeName)
+	}
+
+	fmt.Println("\nWarning: Pods that are 'Not' in 'Running' status")
+	for i := range podNotRunning {
+		fmt.Println(podNotRunning[i].Name, "\t", podNotRunning[i].Status.Phase, "\t", podNotRunning[i].Namespace, "\t", podNotRunning[i].Spec.NodeName)
+	}
+}
+
+func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	pg_stat_code := exec.RunCommandInOperatorPod(context, "ceph", []string{"pg", "stat"}, operatorNamespace, clusterNamespace, false)
+
+	if strings.Contains(pg_stat_code, "active+clean;") {
```

Comment:
right, we need to find a better way. 
We can try using the json parsing with ceph commands but n the other hand, it will put them on either `Info, Warn or Error` mode. We'll have to log for each pg status type.

Or We can just print the ceph command.
thoughts?

For, now I have move the condition checking for `error`  first so that is any of the error state is there is the pg status we say it is `Error`